src/datascience-ui/interactive-common/variableExplorer.tsx

```diff
@@ -169,13 +174,26 @@ export class VariableExplorer extends React.Component<IVariableExplorerProps, IV
         }
     }
 
+    public componentWillReceiveProps(nextProps: IVariableExplorerProps) {
+        this.setState({
+              containerHeight: nextProps.containerHeight,
+              gridHeight: nextProps.gridHeight,
+              variables: nextProps.variables
```

Comment:
If `this.props.variables` changes while a sort is active on the variable explorer's contents, would this cause the variable order to reset to an unordered state? I believe the user would expect their existing sort order to be preserved as variables change.

Reply:
good point, I'll take a look into it